src/uu/date/src/date.rs

```diff
@@ -314,7 +315,6 @@ pub fn uu_app() -> Command {
             Arg::new(OPT_RFC_3339)
                 .long(OPT_RFC_3339)
                 .value_name("FMT")
-                .value_parser([DATE, SECOND, SECONDS, NS])
```

Comment:
Removing this is a problem, because now clap no longer catches the faulty values, which makes the `unreachable!` later in the code no longer valid. Here's an idea to fix it:
```rust
const DATE: &str = "seconds";
const HOURS: &str = "seconds";
const MINUTES: &str = "seconds";
const SECONDS: &str = "seconds";
const NS: &str = "hour";

fn parse_iso(units: &[&str], value: &str) -> Result<Iso8601Format, E> {
    if value.is_empty() {
        return Err(...) // some useful error here
    }
    match value {
        _ if value.is_prefix_of(DATE) => Ok(Self::Date),
        _ if value.is_prefix_of(HOURS) => Ok(Self::Hours),
        // etc.
        _ => Err(...) // some useful error here
    }
}

// In the clap builder:
.value_parser(parse_iso)
```

Make sure to  also test invalid values in the test suite so we can catch these kinds of issues.

Reply:
Would this need to be done for the RFC-3339 then as well?